extensions/amp-analytics/0.1/test/test-custom-browser-events.js

```diff
@@ -0,0 +1,279 @@
+        expect(tracker.root).to.equal(root);
+        expect(tracker.observables_.getHandlerCount()).to.equal(0);
+
+        tracker.add(
+          undefined,
+          AnalyticsEventType.BROWSER_EVENT,
+          changeEventConfig,
+          () => {},
+          false
+        );
+
+        expect(tracker.observables_.getHandlerCount()).to.equal(1);
+
+        tracker.dispose();
+
+        expect(tracker.observables_).to.equal(null);
+      });
+
+      it('should require a selector', () => {
+        allowConsoleError(() => {
+          expect(() => {
+            tracker.add(analyticsElement, AnalyticsEventType.BROWSER_EVENT, {
+              selector: '',
+            });
+          }).to.throw(/Missing required selector on browser event trigger/);
+          expect(() => {
+            tracker.add(analyticsElement, AnalyticsEventType.VIDEO, {
+              selector: [],
+            });
+          }).to.throw(/Missing required selector on browser event trigger/);
+        });
+      });
+
+      it('should error on duplicate selectors', () => {
+        const config = {
+          selector: ['#inputField', '#inputField'],
+        };
+
+        expect(() => {
+          tracker.add(analyticsElement, "blur", config);
+        }).to.throw(
+          /Cannot have duplicate selectors in selectors list: #inputField,#inputField/
+        );
+        expect(() => {
+          tracker.add(analyticsElement, "change", config);
+        }).to.throw(
+          /Cannot have duplicate selectors in selectors list: #inputField,#inputField/
+        );
+      });
+
+      it('fires on one selector with on change', async () => {
+        const fn1 = env.sandbox.stub();
+        const getElementSpy = env.sandbox.spy(root, 'getElement');
+
+        tracker.add(
+          undefined,
+          AnalyticsEventType.BROWSER_EVENT,
+          changeEventConfig,
+          fn1
+        );
+
+        inputField.dispatchEvent(new Event('change'));
+
+        await macroTask();
+        expect(fn1).to.have.callCount(1);
+        expect(fn1).to.be.calledWith(
+          new AnalyticsEvent(inputField, 'change', {})
+        );
+        expect(getElementSpy).to.be.callCount(1);
+        expect(tracker.observables_.handlers_.length).to.equal(1);
+      });
+
+      it('fires on one selector with on blur', async () => {
+        const fn1 = env.sandbox.stub();
+        const getElementSpy = env.sandbox.spy(root, 'getElement');
+
+        tracker.add(
+          undefined,
+          AnalyticsEventType.BROWSER_EVENT,
+          blurEventConfig,
+          fn1
+        );
+
+        inputField.dispatchEvent(new Event('blur'));
+
+        await macroTask();
+        expect(fn1).to.have.callCount(1);
+        expect(fn1).to.be.calledWith(
+          new AnalyticsEvent(inputField, 'blur', {})
+        );
+        expect(getElementSpy).to.be.callCount(1);
+        expect(tracker.observables_.handlers_.length).to.equal(1);
```

Comment:
Typically private state of an object should not be verified in tests. But its ok to leave it for now.